src/services/flatmoney/keeperFee.ts

```diff
@@ -0,0 +1,85 @@
+import { Contract, ethers } from "ethers";
+import { Pool } from "../../entities";
+import IFlatcoinVaultAbi from "../../abi/flatmoney/IFlatcoinVault.json";
+import KeeperFeeAbi from "../../abi/flatmoney/KeeperFee.json";
+import { flatMoneyContractAddresses } from "../../config";
+import BigNumber from "bignumber.js";
+
+export const getKeeperFeeContract = async (pool: Pool): Promise<Contract> => {
+  const flatMoneyContracts = flatMoneyContractAddresses[pool.network];
+  if (!flatMoneyContracts) {
+    throw new Error(
+      `getStableModuleContract: network of ${pool.network} not supported`
+    );
+  }
+  const flatcoinVaultContract = new Contract(
+    flatMoneyContracts.FlatcoinVault,
+    IFlatcoinVaultAbi,
+    pool.signer
+  );
+  const key = ethers.utils.formatBytes32String("keeperFee");
+  const keeperFeeContractAddress: string = await flatcoinVaultContract.callStatic.moduleAddress(
+    key
+  );
+
+  const keeperFeeContract = new ethers.Contract(
+    keeperFeeContractAddress,
+    KeeperFeeAbi,
+    pool.signer
+  );
+  return keeperFeeContract;
+};
+
+export const getKeeperFee = async (
+  pool: Pool,
+  maxKeeperFeeInUsd: number | null
+): Promise<ethers.BigNumber> => {
+  const keeperFeeContract = await getKeeperFeeContract(pool);
+  const blockinfo = await pool.signer.provider.getBlock("latest");
+  const basefee = blockinfo.baseFeePerGas;
+
+  let keeperfee: ethers.BigNumber;
+  if (basefee) {
+    keeperfee = await keeperFeeContract["getKeeperFee(uint256)"](
+      new BigNumber(basefee.toString()).times(2).toFixed(0)
+    );
+  } else {
+    keeperfee = await keeperFeeContract["getKeeperFee()"]();
+  }
+
```

Comment:
Is it this how the keeperfee is calculated in UI? We should just have the same calculation.

Reply:
I checked the frontend code,  they are passing the gasPrice from wagmi to the function and then multiplying with 1.2. I am not sure if the gas price is the base fee from the provider??
https://github.com/dhedge/flatcoin-frontend/blob/0f2830dbf62c1f3da6ba7a8476dea71a3302f48a/libs/flatcoin/page/src/hooks/state/useContractKeeperFee.ts

Reply:
and i think it should be ok to keep the maxKeeperFee check.